src/main/java/org/craftercms/studio/impl/v2/service/cluster/internal/ClusterManagementServiceInternalImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2007-2018 Crafter Software Corporation. All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.craftercms.studio.impl.v2.service.cluster.internal;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.craftercms.studio.api.v2.dal.ClusterDAO;
+import org.craftercms.studio.api.v2.dal.ClusterMember;
+import org.craftercms.studio.api.v2.service.cluster.internal.ClusterManagementServiceInternal;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class ClusterManagementServiceInternalImpl implements ClusterManagementServiceInternal {
+
+    private ClusterDAO clusterDao;
+
+    @Override
+    public List<ClusterMember> getAllMembers() {
+        return clusterDao.getAllMembers();
+    }
+
+    @Override
+    public boolean removeMembers(List<Long> memberIds) {
+        if (CollectionUtils.isNotEmpty(memberIds)) {
+            Map<String, Object> params = new HashMap<String, Object>();
+            params.put("memberIds", memberIds);
+            int result = clusterDao.removeMembers(params);
+            return result > 0;
+        } else {
+            return true;
+        }
+    }
+
+    @Override
+    public boolean addMember(ClusterMember clusterMember) {
+        if (StringUtils.isEmpty(clusterMember.getClusterId())) {
+            clusterMember.setClusterId(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberName())) {
+            clusterMember.setClusterMemberName(UUID.randomUUID().toString());
+        }
+        if (StringUtils.isEmpty(clusterMember.getClusterMemberIp())) {
+            clusterMember.setClusterMemberIp("127.0.0.1");
```

Comment:
Why?
Shouldn't this cause an error?

Reply:
Difference between DB model and API model. We need to discuss it. For now, that code is there not to break service